2020-09-12 08:22:00,453 ERROR: Exception on /explore [GET] [in /Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/michellewong/Documents/Speakup/app/main/routes.py", line 148, in explore
    next_url=next_url, prev_url=prev_url, pages=pages)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/templating.py", line 138, in render_template
    ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 930, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 883, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 857, in _load_template
    template = self.loader.load(self, name, globals)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/loaders.py", line 127, in load
    code = environment.compile(source, name, filename)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 638, in compile
    self.handle_exception(source=source_hint)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/michellewong/Documents/Speakup/app/templates/explore.html", line 51, in template
    {% for x in range(1,{{page }}): %}
jinja2.exceptions.TemplateSyntaxError: expected token ':', got '}'
2020-09-12 08:22:24,142 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:22:26,986 ERROR: Exception on /explore [GET] [in /Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py:1892]
Traceback (most recent call last):
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/Users/michellewong/Documents/Speakup/app/main/routes.py", line 148, in explore
    next_url=next_url, prev_url=prev_url, pages=pages)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 1090, in render
    self.environment.handle_exception()
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/Users/michellewong/Documents/Speakup/venv/lib/python3.7/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/Users/michellewong/Documents/Speakup/app/templates/explore.html", line 1, in top-level template code
    {% extends "navbar.html" %}
  File "/Users/michellewong/Documents/Speakup/app/templates/navbar.html", line 114, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/michellewong/Documents/Speakup/app/templates/explore.html", line 51, in block "content"
    {% for x in range(1,page): %}
TypeError: 'Undefined' object cannot be interpreted as an integer
2020-09-12 08:22:53,358 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:23:10,519 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:25:19,698 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:27:37,489 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:28:01,412 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:28:17,293 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:31:58,390 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:34:20,764 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:40:57,107 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:47:25,585 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:49:36,429 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:51:45,674 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:53:18,018 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:54:11,246 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:54:38,613 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:56:00,160 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:56:28,950 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:58:14,639 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:59:08,772 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 08:59:30,660 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:09:11,823 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:10:05,292 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:11:27,467 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:11:47,424 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:12:12,658 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:12:49,671 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:13:10,883 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:13:42,912 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:13:58,792 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:15:34,818 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:16:25,698 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:16:51,267 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:19:32,180 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:28:41,898 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:34:53,740 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:35:23,726 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:41:35,027 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
2020-09-12 09:45:27,775 INFO: Microblog startup [in /Users/michellewong/Documents/Speakup/app/__init__.py:90]
